lib/util/widgets/switch_widget.dart

```diff
@@ -1,41 +1,33 @@
 import 'package:flutter/material.dart';
 
 class SwitchWidget extends StatelessWidget {
-  final Function(bool) onThemeChanged;
-  final Function(bool) onDemoModeChanged;
-  final bool isDarkMode;
-  final bool isDemoMode;
+  final Function(bool) onChanged;
+  final bool isActive;
 
   const SwitchWidget({
     super.key,
-    required this.onThemeChanged,
-    required this.onDemoModeChanged,
-    required this.isDarkMode,
-    required this.isDemoMode,
+    required this.onChanged,
+    required this.isActive,
   });
 
   @override
   Widget build(BuildContext context) {
     return GestureDetector(
       onTap: () {
-        if (isDemoMode) {
-          onDemoModeChanged(!isDemoMode);
-        } else {
-          onThemeChanged(!isDarkMode);
-        }
+        onChanged(!isActive);
       },
       child: AnimatedContainer(
         duration: const Duration(milliseconds: 150),
         width: 50.0,
         height: 30.0,
         decoration: BoxDecoration(
           borderRadius: BorderRadius.circular(15.0),
-          color: (isDarkMode || isDemoMode)
+          color: isActive
               ? Theme.of(context).colorScheme.tertiaryContainer
               : Theme.of(context).colorScheme.background,
```

Comment:
border: isActive
                ? null
                : Border.all(
              color: Theme.of(context).colorScheme.outline,)